tests/swin/test_modeling_swin.py

```diff
@@ -307,6 +309,78 @@ def test_attention_outputs(self):
                     [self.model_tester.num_heads[0], window_size_squared, window_size_squared],
                 )
 
+    def test_model_outputs_equivalence(self):
+        config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
+
+        def set_nan_tensor_to_zero(t):
+
+            t[t != t] = 0
+            return t
+
+        def check_equivalence(model, tuple_inputs, dict_inputs, additional_kwargs={}):
+            with torch.no_grad():
+                tuple_output = model(**tuple_inputs, return_dict=False, **additional_kwargs)
+                dict_output = model(**dict_inputs, return_dict=True, **additional_kwargs).to_tuple()
+
+                def recursive_check(tuple_object, dict_object):
+                    # for spatial dimensions and exit condition for recursion
+                    if type(tuple_object) is int:
```

Comment:
You wouldn't need this extra check if you had stuck with tensors inside the model outputs

Reply:
I was not aware all the outputs must be tensors

Reply:
By the Trainer.

Reply:
they must not be reduced by the Trainer, they are batch dependent.

Reply:
So they shouldn't be in this output.

Reply:
Removed in favor or `reshaped_hidden_states`